Breaking change in ClinicalTrials API base URL

* pytrials version: 0.2.0
* Python version: 3.10.4
* Operating System: MacOS Ventura 13.4

### Description

Within the past week, I have been getting the following error when instantiating `ClinicalTrials`:
```
tests/test_client.py:7: in <module>
    ct = ClinicalTrials()
pytrials/client.py:24: in __init__
    self.api_info = self.__api_info()
pytrials/client.py:35: in __api_info
    last_updated = json_handler(
pytrials/utils.py:21: in json_handler
    return request_ct(url).json()
env/lib/python3.10/site-packages/requests/models.py:975: in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
E   requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

The error occurs both in my code and in the `pytrials` tests. After some investigation, it appears that the
`https://clinicaltrials.gov/api/info/data_vrs?fmt=json`
endpoint is no longer valid, causing a failure upon class instantiation, when `self.__api_info` queries for the API version.

### What I Did

I'm opening two PRs:
- One to change the `BASE_URL` to `https://classic.clinicaltrials.gov/api/info/data_vrs?fmt=json`, which appears to be undocumented but it's what the examples in the ClinicalTrials API docs use now.
- Raise for error if response status is not 200.

--- a/pytrials/client.py
+++ b/pytrials/client.py
@@ -14,7 +14,7 @@ class ClinicalTrials:
         time the database was updated.
     """
 
-    _BASE_URL = "https://clinicaltrials.gov/api/"
+    _BASE_URL = "https://classic.clinicaltrials.gov/api/"
     _INFO = "info/"
     _QUERY = "query/"
     _JSON = "fmt=json"

